::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
: Copyright (c) 2017 Mirantis Inc., Enea AB and others.
:
: All rights reserved. This program and the accompanying materials
: are made available under the terms of the Apache License, Version 2.0
: which accompanies this distribution, and is available at
: http://www.apache.org/licenses/LICENSE-2.0
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Fri, 27 Oct 2017 21:00:03 +0200
Subject: [PATCH] Use PXE interface index from net_config

Instead of assuming the PXE network interface is always the first,
read its index from 'net_config' section.

Change-Id: I0b4e9fc89297f4bd718a2e5336746917347c89e1
Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 config/installers/fuel/pod_config.yml.j2 | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/config/installers/fuel/pod_config.yml.j2 b/config/installers/fuel/pod_config.yml.j2
index 890b380..abded17 100644
--- a/config/installers/fuel/pod_config.yml.j2
+++ b/config/installers/fuel/pod_config.yml.j2
@@ -8,6 +8,7 @@
     {%- if conf.net_config.public.dns is defined -%}
         {%- set dns_public = conf['net_config']['public']['dns'] -%}
     {%- endif -%}
+    {%- set pxe_interface = conf['net_config']['admin']['interface'] -%}
 {%- else -%}
     {%- set net_admin = '192.168.11.0' -%}
     {%- set net_mgmt = '172.16.10.0' -%}
@@ -15,13 +16,12 @@
     {%- set net_public = '172.30.10.0' -%}
     {%- set vlan_mgmt = '300' -%}
     {%- set vlan_private = '1000' -%}
+    {%- set pxe_interface = 0 -%}
 {%- endif -%}

 {%- if dns_public is not defined -%}
     {%- set dns_public = [ '8.8.8.8', '8.8.4.4' ] -%}
 {%- endif -%}
-
-{%- set pxe_interface = 0 -%}
 ---
 parameters:
   _param:
